src/main/java/org/sonarsource/sonarlint/ls/http/ApacheHttpClient.java

```diff
@@ -99,33 +96,47 @@ private Response executeSync(SimpleHttpRequest httpRequest) {
     }
   }
 
-  private CompletableFuture<Response> executeAsync(SimpleHttpRequest httpRequest) {
-    if (login != null) {
-      httpRequest.setHeader("Authorization", basic(login, password == null ? "" : password));
+  private CompletableFuture<Response> executeAsync(SimpleRequestBuilder httpRequest) {
+    if (token != null) {
+      httpRequest.setHeader(HttpHeaders.AUTHORIZATION, basic(token, ""));
     }
-    CompletableFuture<Response> futureResponse = new CompletableFuture<>();
-    Future<SimpleHttpResponse> httpFuture = client.execute(httpRequest, new FutureCallback<SimpleHttpResponse>() {
+    CompletableFutureWrapper futureWrapper = new CompletableFutureWrapper(httpRequest);
+    Future<SimpleHttpResponse> httpFuture = client.execute(httpRequest.build(), futureWrapper);
+    futureWrapper.wrapped = httpFuture;
+    return futureWrapper;
+  }
 
-      @Override
-      public void completed(SimpleHttpResponse result) {
-        futureResponse.complete(new ApacheHttpResponse(httpRequest.getRequestUri(), result));
-      }
+  private static final class CompletableFutureWrapper extends CompletableFuture<Response> implements FutureCallback<SimpleHttpResponse> {
 
-      @Override
-      public void failed(Exception ex) {
-        futureResponse.completeExceptionally(ex);
-      }
+    private Future<SimpleHttpResponse> wrapped;
+    private final SimpleRequestBuilder httpRequest;
 
-      @Override
-      public void cancelled() {
-        // nothing to do, the completable future is already canceled
-      }
-    });
-    return futureResponse.whenComplete((r, t) -> {
-      if (t instanceof CancellationException) {
-        httpFuture.cancel(false);
```

Comment:
@damien-urruty-sonarsource I noticed that this code was never called and the original Future was never cancelled. Since I copied this code from SLI, we might want to fix it there as well (or at least create a ticket).